test/hotspot/jtreg/compiler/c2/TestPropagateTypeRefinementToIGVN.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8359602
+ * @summary The control input of a ModI node should be discarded if it is
+ *          possible to prove that the divisor can never be 0.
+ *          VerifyIterativeGVN checks that this optimization was applied
+ * @requires vm.debug == true
```

Comment:
Is that really required? Can't you get away with a `-XX:+IgnoreUnrecognizedVMOptions` if the problem is unrecognized VM options? Would be nice to know if it works in product too.

Reply:
The entire `verify_PhaseIterGVN()` call hides behind a `NOT_PRODUCT` macro, so unless I am missing something fundemental here I think we really need the debug mode. You can find the call [here](https://github.com/benoitmaillard/jdk/blob/2ac5e725e472578618c5ab47425c52843dc119bb/src/hotspot/share/opto/phaseX.cpp#L1069).